web/src/components/dashboard/HideableTasks.test.tsx

```diff
@@ -0,0 +1,85 @@
+import { getMergedConfig } from "@/contexts/configContext";
+import { generatePreferences, generateProfileData, generateTask, generateUserData } from "@/test/factories";
+import { randomElementFromArray } from "@/test/helpers";
+import { useMockRoadmap } from "@/test/mock/mockUseRoadmap";
+import { useMockUserData } from "@/test/mock/mockUseUserData";
+import { setupStatefulUserDataContext, WithStatefulUserData } from "@/test/mock/withStatefulUserData";
+import { OperatingPhases } from "@businessnjgovnavigator/shared/operatingPhase";
+import { UserData } from "@businessnjgovnavigator/shared/userData";
+import { fireEvent, render, screen } from "@testing-library/react";
+import { HideableTasks } from "./HideableTasks";
+
+jest.mock("@/lib/data-hooks/useUserData", () => ({ useUserData: jest.fn() }));
+jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
+
+const Config = getMergedConfig();
+
+describe("HideableTasks", () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    setupStatefulUserDataContext();
+    useMockRoadmap({});
+  });
+
+  const renderHideableTask = (initialUserData?: UserData) =>
+    render(
+      <WithStatefulUserData initialUserData={initialUserData}>
+        <HideableTasks />
+      </WithStatefulUserData>
+    );
+
+  it("does not render HideableTasks for operating phases that don't display HideableRoadmapTasks", () => {
+    const randomOperatingPhase = randomElementFromArray(
+      OperatingPhases.filter((obj) => obj.displayHideableRoadmapTasks !== true)
+    );
+    useMockUserData({ profileData: generateProfileData({ operatingPhase: randomOperatingPhase.id }) });
```

Comment:
As a heads up, you're using 2 different mocking patterns for userData in the same test here. It works, but it's a little confusing to read.

Your render function is expecting initialUserData, and using the <WithStatefulUserData> wrapper to set up a mocked _dynamic_ userData hook. We use this anytime we want to test that changing the state of the userData changes something on the page. You're right that you need this stateful wrapped, because the open/close is controlled by the userData preferences, and without this, then preferences wouldn't actually change and the component wouldn't work.

Whereas `useMockUserData` is for _static_ mocking. It only mocks what happens when we ask for userData. We typically use this when we just need to start with some value but don't ever change it.

Using both of them in the same test is a little confusing and hard-to-follow because you have to think harder about how the two different mocking patterns interact to ensure it actually reflects reality.

I'd recommend removing `useMockUserData` altogether, because we do need the dynamic mocking here. And you can pass the information you have defined here into your render function as initialUserData